internal/databases/mongo/binary/restore.go

```diff
@@ -0,0 +1,80 @@
+package binary
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/utility"
+)
+
+type RestoreService struct {
+	Context       context.Context
+	LocalStorage  *LocalStorage
+	BackupStorage *BackupStorage
+}
+
+func CreateRestoreService(
+	ctx context.Context,
+	localStorage *LocalStorage,
+	backupStorage *BackupStorage) (*RestoreService, error) {
+
+	return &RestoreService{
+		Context:       ctx,
+		LocalStorage:  localStorage,
+		BackupStorage: backupStorage,
+	}, nil
+}
+
+func (restoreService *RestoreService) DoRestore(restoreMongodVersion string) error {
+	mongodBackupMeta, err := restoreService.BackupStorage.DownloadMongodBackupMeta()
+	if err != nil {
+		return err
+	}
+
+	err = EnsureCompatibilityToRestoreMongodVersions(restoreMongodVersion, mongodBackupMeta.MongodMeta.Version)
+	if err != nil {
+		return err
+	}
+
+	err = restoreService.LocalStorage.EnsureMongodFsLockFileIsEmpty()
+	if err != nil {
+		return err
+	}
+
+	err = restoreService.LocalStorage.CleanupMongodDbPath()
+	if err != nil {
+		return err
+	}
+
+	tracelog.InfoLogger.Println("Download backup files to dbPath")
+	return restoreService.DownloadFilesFromBackup(mongodBackupMeta)
+}
+
+func (restoreService *RestoreService) DownloadFilesFromBackup(backupMeta *MongodBackupMeta) error {
+	err := restoreService.LocalStorage.CreateDirectories(backupMeta.BackupDirectories)
```

Comment:
Maybe add a check that the target directory is empty to avoid some unexpected results?